### PR TITLE
Tox smoketest for scan scripts

### PR DESCRIPTION
**Pull request by g-k**
refs: https://github.com/zaproxy/zaproxy/pull/3746#discussion_r128315445

Initial/WIP PR for smoketesting the docker scan scripts blocked on how to exit with code 0.

Changes:
* zap-api-scan python3 compatibility fix
* tox file for making sure the scripts run (run `tox` from `build/docker`)

Questions:

* currently this fails with exit code 3. Possible solutions:
  * test against an actual API or app (probably slow but good to test)
  * have tox ignore the exit code (makes the tests useless)
  * add an option (like `--version`, `--usage`, `--help` or a shorthand version)  to the scan scripts that exits with 0
* should we run this in CI?

### CODE REVIEW

File: build/docker/tox.ini [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[tox]
+skipsdist = True
+envlist = py26,py27,py34,py35,py36
+[testenv]
+deps = -rrequirements.txt
```
Comment (thc202) [inline build/docker/tox.ini]: It's missing the `requirements.txt` file.

**Comment by thc202** [comment]
I guess we could start with the 3rd solution?

Worth including that if it doesn't add much time to the build(?).

**Comment by g-k** [comment]
Updated to add a `-h` option for help and a requirements file.

The whitespace changes could go in a separate PR if that'd be preferable.

This still isn't running from CI. How do we want to do that?

Should we:

1. run tox from ant
1. run it from travis CI
1. run it from Circle CI?
1. something else

I'm leaning toward 1. but haven't looked into how to do that. Should that be a separate PR?

**Comment by kingthorin** [comment]
@g-k can you address the conflicts?

**Comment by g-k** [comment]
@kingthorin yep, rebased on develop.

File: build/docker/zap-api-scan.py [hunk 2]
```diff
@@ -244,13 +247,10 @@ def main(argv):
     elif config_url:
         # load config file from url
         try:
-            load_config(urllib2.urlopen(config_url), config_dict, config_msg, out_of_scope_dict)
+            load_config(urlopen(config_url), config_dict, config_msg, out_of_scope_dict)
         except ValueError as e:
             logging.warning(e)
             sys.exit(3)
-        except:
```
Comment (thc202) [inline build/docker/zap-api-scan.py]: Was this removed on purpose?

**Comment by thc202** [comment]
I think we can leave the CI changes to other PR (in the meantime we can run it manually, the scripts don't change that often anyway). Regarding the option, I guess we can go with 2? (We can run tox directly without Ant, any advantages for the latter?)

File: build/docker/zap-api-scan.py (hunk 2 above)
Comment (g-k) [reply build/docker/zap-api-scan.py]: No, I got confused resolving a merge conflict.

Comment (g-k) [reply build/docker/zap-api-scan.py]: Fixed.

**Comment by g-k** [comment]
> I think we can leave the CI changes to other PR

👍 

> I guess we can go with 2? (We can run tox directly without Ant, any advantages for the latter (i.e. run with Ant)?)

I guess it's slightly simpler, but would make tests slower.

2. sounds good.

**Comment by thc202** [comment]
Thanks @g-k and @kingthorin for bumping :)

**Pull request merged**